src/Feature/FeatureTrace.hs

```diff
@@ -47,6 +47,10 @@ simplifyFeatureTraceOfNodes f t d = \v -> if member v d then simplifyFeatureTrac
 Calculates the presence condition of a node (third argument) in the given tree (first argument) with the given feature traces (second argument).
 See Equation 1 in the paper.
 -}
+-- TODO(benjamin) In the paper `trace node` is not included if `node` is
+-- `Mandatory`. Does this code need the following invariant?
```

Comment:
That's the reason why the definition of `pc` is split into `pc` and `pcParentPart`. If I want to get the presence condition of a mandatory node, its own feature mapping `trace n` has to be included although it is actually always `Nothing`. Given that the own feature mapping `trace n` thus has to and can be included for all nodes, I just add it always. You are right though, that it could directly be simplified in case the node is mandatory.